Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -527,6 +527,19 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
         m_stringBuilder.append(m_indent, "}\n\n"_s);
     }
 
+    if (m_shaderModule.usesFtoi()) {
+        m_stringBuilder.append(m_indent, "template <typename T, typename S>\n"_s,
+            m_indent, "static T __wgslFtoi(S value)\n"_s,
+            m_indent, "{\n"_s);
+        {
+            IndentationScope scope(m_indent);
+            m_stringBuilder.append(m_indent, "auto min = metal::numeric_limits<T>::min();\n"_s,
+                m_indent, "auto max = metal::numeric_limits<T>::max();\n"_s,
```

Comment:
Oh I just realized these are `metal::numeric_limits` variants, please ignore the earlier comment

Reply:
It's also operating on the target type, which is the integer type, so I think it's ok. But it won't handle booleans right now, so I have to fix that

Reply:
Yes this is great, sorry ignore my earlier comment

Reply:
I implemented it for booleans too, but then I checked and it doesn't seem like it's UB to convert any float values to bool. To confirm I tested many different values with UBSan and no issues were raised, so I'll leave it out for now.